eblocker-icapserver/src/main/java/org/eblocker/server/common/blacklist/Cache.java

```diff
@@ -185,32 +180,12 @@ private boolean checkIndexFile() throws IOException {
         }
         JsonNode cacheFormatField = node.get("format");
         if (cacheFormatField == null || cacheFormatField.intValue() < CacheIndex.CACHE_FORMAT) {
-            upgradeIndex(node);
+            cacheIndexRepository.upgradeJsonFile(node);
+            // remove obsolete profile filters
+            deleteFilesInDirectory(PROFILES_DIR);
         }
 
         return true;
     }
 
-    private void upgradeIndex(@Nonnull JsonNode node) throws IOException {
-        // update filter meta data
-        ObjectReader filtersReader = objectMapper.readerFor(new TypeReference<Map<Integer, List<CachedFileFilter>>>() {
-        });
-
-        Map<Integer, List<CachedFileFilter>> cachedFilters = filtersReader.readValue(node.get("filters"));
-        Map<Integer, List<CachedFileFilter>> updatedFilters = cachedFilters.values().stream()
-                .flatMap(Collection::stream)
-                .map(filter -> {
-                    String format = filter.getFileFilterFileName() != null ? "domainblacklist/string" : "domainblacklist/bloom";
-                    return new CachedFileFilter(filter.getKey(), filter.getBloomFilterFileName(), filter.getFileFilterFileName(), format, filter.isDeleted());
-                })
-                .sorted(Comparator.comparing(filter -> filter.getKey().getVersion(), Comparator.reverseOrder()))
```

Comment:
I have moved this method to the repo class and I have also moved the sorting from here to the constructor of the CacheIndex. This means that it is no longer possible to set unsorted values via this constructor.